### PR TITLE
[Issue-1171] - check tenancy on invalidate content jobs if tenancy is turned on

### LINKED ISSUE #1171: Creating an invalidate content request (purge) on a delivery service should check tenancy (if turned on)
The following API route is used to create an "Invalidate content" (aka purge request) on a delivery service:

$r->post("/api/$version/user/current/jobs")->over( authenticated => 1, not_ldap => 1 )->to( 'Job#create_current_user_job', namespace => $namespace );

This route current checks:

if role is ops (or higher) OR if delivery service is assigned to user.

instead it should follow this logic:

if (use_tenancy = 1) { check tenancy on delivery service }
else { check if role is ops (or higher) OR if delivery service is assigned to user }

### PR DESCRIPTION
**Pull request by mitchell852**
fixes #1171

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/274/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/276/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/232/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/234/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/277/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/235/

File: traffic_ops/app/lib/API/Job.pm [hunk 1]
```diff
@@ -148,18 +164,21 @@ sub create_current_user_job {
...
 
-		# select deliveryservice from deliveryservice_tmuser where deliveryservice=$ds_id
-		my $dbh = $self->db->resultset('DeliveryserviceTmuser')->search( { deliveryservice => $ds_id, tm_user_id => $tm_user_id }, { id => 1 } );
-		my $count = $dbh->count();
+	my $tenant_utils = Utils::Tenant->new($self);
+	my $tenants_data = $tenant_utils->create_tenants_data_from_db();
 
-		if ( $count == 0 ) {
-			return $self->forbidden("Forbidden. Delivery service not assigned to user.");
+	if ( $tenant_utils->use_tenancy) {
```
Comment (nir-sopher) [inline traffic_ops/app/lib/API/Job.pm]: Again, missing "()"

File: traffic_ops/app/lib/API/Job.pm [hunk 2]
```diff
@@ -41,12 +42,27 @@ sub index {
...
+
+		my $ds = $self->db->resultset('Deliveryservice')->find( { id => $ds_id } );
+		if ( !defined($ds) ) {
+			return $self->not_found();
+		}
+
+		my $tenant_utils = Utils::Tenant->new($self);
+		my $tenants_data = $tenant_utils->create_tenants_data_from_db();
+
+		if ( $tenant_utils->use_tenancy) {
```
Comment (nir-sopher) [inline traffic_ops/app/lib/API/Job.pm]: use_tenancy is a function, you are missing the "()", unless it is not required in perl

Comment (mitchell852) [reply traffic_ops/app/lib/API/Job.pm]: good catch. thanks. but i tested it so maybe perl doesn't care?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/296/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/255/

**Pull request merged**